susemanager-utils/susemanager-sls/modules/engines/mgr_events.py

```diff
@@ -127,13 +128,16 @@ def _insert(self, tag, data):
             fnmatch.fnmatch(tag, "suse/manager/image_deployed"),
             fnmatch.fnmatch(tag, "suse/systemid/generate")
         ]) and not self._is_salt_mine_event(tag, data) and not self._is_presence_ping(tag, data):
-            log.debug("%s: Adding event to queue -> %s", __name__, tag)
+            queue = int(hashlib.md5(data.get("id").encode(self.connection.encoding)).hexdigest()[0:8], 16) \
+                      % self.config['events']['thread_pool_size'] \
+                    if data.get("id") is not None else -1
```

Comment:
How about using 0 to thread_pool_size as the range, instead of -1 to thread_pool_size -1? That should make code on the Java side a bit more readable because that can be used directly as an array or list index.